File: e-commerce-backend/src/main/java/com/TechKun/model/SupportTicket.java
package com.TechKun.model;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Entity
@Data
@Table(name = "SupportTicket")
public class SupportTicket {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ticket_id", nullable = false)
    private Integer ticketId;

    @Column(name = "subject", nullable = false)
    private String subject;

    @Column(name = "description", nullable = false, columnDefinition = "TEXT")
    private String description;

    @Column(name = "status", nullable = false)
    private String status;

    @ManyToOne
    @JoinColumn(name = "customer_id", nullable = false)
    private ShopUser customer;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}

File: e-commerce-frontend/src/types/domains/support_ticket.ts
import { CustomerContact, ShopUser } from "./user";

export interface SupportTicket {
    ticketId: number;
    subject: string;
    description: string;
    status: string;
    customer: ShopUser;
    createdAt: Date;
}

export interface SupportTicketDTO {
    subject: string;
    description: string;
    status: string;
}

export interface SupportTicketDetails {
    ticketId: number;
    subject: string;
    description: string;
    status: string;
    customer: CustomerContact;
}

export interface SupportTicketQueryOptions {
    status: string;
    customerName: string;
    fromDate: Date;
    toDate: Date;
}